### DISCUSSION TITLE
Error in creating MNW2 object

### DISCUSSION THREAD

**Discussion by kimlee100** [modflowpy/flopy#1541] [Q&A] [answered]
Hello,

I rewriting MNW2 package by modifying some wells. Following the MNW2 example, I converted to numpy array both node_data and stress_period_data, however when I try to create MNW2 object it is giving an error that node_data has no dtype. Checked data types and they are quite similar to the ones in the example. Can you suggest why it is not creating an object?

# Error message
----> 1 mnw2modif = flopy.modflow.ModflowMnw2(model=mnwmodif, node_data=node_data, stress_period_data=stress_period_data)

C:\ProgramData\Anaconda3\lib\site-packages\flopy\modflow\mfmnw2.py in __init__(self, model, mnwmax, nodtot, ipakcb, mnwprnt, aux, node_data, mnw, stress_period_data, itmp, extension, unitnumber, filenames, gwt)
   1060             names = [
   1061                 n
-> 1062                 for n in node_data.dtype.names
   1063                 if n in self.node_data.dtype.names
   1064             ]

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\generic.py in __getattr__(self, name)
   5463             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5464                 return self[name]
-> 5465             return object.__getattribute__(self, name)
   5466 
   5467     def __setattr__(self, name: str, value) -> None:

AttributeError: 'DataFrame' object has no attribute 'dtype'

**Comment by p-ortega** [comment, marked as answer]
You have to convert the node data to a recarray. From the error node_data is a dataframe so do the following:

`node_data = node_data.to_records()`
